File: trading-ui/jgod-trading-ui/src/types/doctrineAlert.ts
/**
 * Doctrine Alert Types
 * 
 * TypeScript type definitions for Doctrine alerts.
 */

export type DoctrineAlertSeverity = "info" | "warning" | "critical";

export type DoctrineAlertSource = "position" | "prediction" | "conflict" | "error";

export type DoctrineRef = {
  book_id: string;
  section_id: string;
  rule_id?: string | null;
};

export type DoctrineAlertItem = {
  id: string;
  symbol: string;
  name?: string | null;
  severity: DoctrineAlertSeverity;
  source: DoctrineAlertSource;
  title: string;
  message: string;
  metric_name: string;
  metric_value?: number | null;
  threshold?: number | null;
  conflict_score?: number | null;
  consensus_score?: number | null;
  final_score?: number | null;
  raw_score?: number | null;
  doctrine_refs: DoctrineRef[];
  tags: string[];
  created_at: string;
};

File: jgod/api/schemas/doctrine_alert.py
"""Doctrine Alert API Schemas

Pydantic models for Doctrine Alert API endpoints.
"""

from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field

from jgod.doctrine_alert.models import (
    DoctrineAlertSeverity,
    DoctrineAlertSource,
    DoctrineRef,
    DoctrineAlertItem as DoctrineAlertItemModel,
)


# Re-export models as API schemas
DoctrineAlertItem = DoctrineAlertItemModel


class DoctrineAlertSummary(BaseModel):
    """Summary of Doctrine alerts"""
    total_by_severity: dict[str, int] = Field(default_factory=dict)  # {"critical": 5, "warning": 10, "info": 2}
    total_by_source: dict[str, int] = Field(default_factory=dict)  # {"position": 3, "prediction": 5, "conflict": 9}
    total: int = 0

